### PR TITLE
demo of a working pg database

### PR DESCRIPTION
**Pull request by scottPlusPlus**
Adds a demo postgres database.
Runs a test that just updates a counter.  Wait for the instance to reset (or force it to rebuild) and note that the counter is persisted.

REQUIRES that the Postgres db be setup on the heroku app:
https://devcenter.heroku.com/articles/heroku-postgresql
```
heroku addons:create heroku-postgresql:hobby-dev
```

### CODE REVIEW

**Review by ArendPeter — CHANGES_REQUESTED** [review]
Aside from the conflicts, it looks good to me!

File: backend/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import express from 'express';
 import { testMockUserStore } from './auth/test/TestMockUserStore';
 import StarResults from './StarResults.cjs';
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (scottPlusPlus) [reply backend/src/index.ts]: whoops.  okay, I think I fixed it.

[the remark this answers is not in the document]
Comment (mikefranze) [reply backend/src/index.ts]: Looks good to me

**Review by ArendPeter — APPROVED** [review]
Looks great!

**Comment by mikefranze** [comment]
Looks like it errored in the build process in heroku 
> tsc --project ./

internal/modules/cjs/loader.js:905
  throw err;
  ^

Error: Cannot find module 'pg'
Require stack:
- /app/backend/build/backend/src/test/DBTest.js
- /app/backend/build/backend/src/test/TempTestSuite.js
- /app/backend/build/backend/src/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/app/backend/build/backend/src/test/DBTest.js:15:18)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/app/backend/build/backend/src/test/DBTest.js',
    '/app/backend/build/backend/src/test/TempTestSuite.js',
    '/app/backend/build/backend/src/index.js'
  ]
}

**Comment by mikefranze** [comment]
pg needed to be added to dependencies list. How do you run this test to see that it works?

**Comment by mikefranze** [comment]
Never mind, figured it out and it works!

**Pull request merged**